Library/Homebrew/api.rb

```diff
@@ -92,21 +92,62 @@ def self.fetch_json_api_file(endpoint, target:)
 
         retry
       end
+
+      if jws
+        success, data = verify_and_parse_jws(json_data)
+        unless success
+          target.unlink
+          odie "Failed to verify integrity of #{url} (#{data})."
+        end
+        [data, !skip_download]
+      else
+        [json_data, !skip_download]
+      end
+    end
+
+    sig { params(json_data: Hash).returns([T::Boolean, T.any(String, Array, Hash)]) }
+    def self.verify_and_parse_jws(json_data)
+      signatures = json_data["signatures"]
+      homebrew_signature = signatures&.find { |sig| sig.dig("header", "kid") == "homebrew-1" }
+      return false, "key not found" if homebrew_signature.nil?
```

Comment:
Might want to say what key isn't found. I'm not sure the value to users of these specific messages unless we make clear they are for debugging.

Reply:
It is debugging basically. The user should never see them in normal use unless they are victim to MITM or we roll the key in the future, but equally we _must_ abort if it happens.